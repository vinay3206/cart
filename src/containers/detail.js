import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';

import Header from '../components/header';
import ProductDetails from '../components/product-detail';
import { fetchProductDetails, addToCart } from '../actions/product';

export class Details extends Component {
  componentWillMount() {
    const { match: { params: { id } } } = this.props;
    const { fetchProductDetails } = this.props;
    // making call to fetch the product details here
    fetchProductDetails(id);
  }


  render() {
    const { productDetail, addToCart, cart } = this.props;
    // calculating the total items in the cart
    const cartItemsCount = cart.items
      .reduce((total, { quantity }) => total + quantity, 0);

    return (
        <div>
          <Header showHomeLink showCartLink cartItemsCount={cartItemsCount}/>
          <div className="flex-container">
            <div className="home-content-container flex-6">
              { productDetail && <ProductDetails product={productDetail} onAddToCart={addToCart}/> }
            </div>
          </div>
        </div>
    );
  }
}

Details.propTypes = {
  match: PropTypes.object.isRequired,
  fetchProductDetails: PropTypes.func.isRequired,
  productDetail: PropTypes.object.isRequired,
  addToCart: PropTypes.func.isRequired,
  cart: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    productDetail: state.product.productDetail,
    cart: state.cart,
  };
}


const mapDispatchToProps = (dispatch) =>
  bindActionCreators({ fetchProductDetails, addToCart }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Details);
