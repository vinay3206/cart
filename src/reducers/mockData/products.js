const imageUrl = 'https://4.bp.blogspot.com/-aI_KMIohZHQ/VN_63b-lsQI/AAAAAAAAZd4/6sifxckbEgQ/s1600/F%26N%2BNutriTea%2BBarley%2BDrink_1L.jpg';

const products = [
  {
    id: 120,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Nutri Well Barly',
    price: 2.25,
    currency: '$',
    imageUrl,
    brand: {
      id: 1,
      label: 'Mangolia',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 121,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Nutri Well Lemon Grass with Ginger',
    price: 2.35,
    currency: '$',
    imageUrl,
    brand: {
      id: 2,
      label: 'Meiji',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 122,
    capacity: 500,
    capacityUnit: 'ml',
    title: 'Nutri Well Water Chestnut and Sugar cane',
    price: 1.25,
    currency: '$',
    imageUrl,
    brand: {
      id: 2,
      label: 'Meiji',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 123,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Nutri Well Water cherysenthamums with Wolfberry',
    price: 2.25,
    currency: '$',
    imageUrl,
    brand: {
      id: 1,
      label: 'Mangolia',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 124,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Nutri Uji Cha Yuzu Green Tea',
    price: 5.05,
    currency: '$',
    imageUrl,
    brand: {
      id: 1,
      label: 'Mangolia',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
  {
    id: 125,
    capacity: 1,
    capacityUnit: 'L',
    title: 'Marigold Uji Cha Momo Green Tea',
    price: 7.25,
    currency: '$',
    imageUrl,
    brand: {
      id: 2,
      label: 'Meiji',
    },
    description: 'F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting',
  },
];

export default products;
