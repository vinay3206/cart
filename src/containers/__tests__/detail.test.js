import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import { Details } from '../detail';
import products from '../../reducers/mockData/products';

const product = products[0];

const cartItems = products.map((product) => {
  const returnObj = {
    qunatity: 1,
    item: product,
  };
  return returnObj;
});

const cart = { items: cartItems };

const match = { params: { id: 120 } };

it('renders correctly', () => {
  const component = <Details
    match={match}
    productDetail={product}
    fetchProductDetails={() => {}}
    addToCart={() => {}}
    cart={cart}
  />;
  const tree = renderer
    .create(<MemoryRouter>{component}</MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
