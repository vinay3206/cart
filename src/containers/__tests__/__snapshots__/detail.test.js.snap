// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="pull-right"
    >
      <a
        className="btn-flat btn-blue-light"
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="btn-flat btn-red"
        href="/cart"
        onClick={[Function]}
      >
        Cart
        
      </a>
    </div>
  </div>
  <div
    className="flex-container"
  >
    <div
      className="home-content-container flex-6"
    >
      <div
        className="product-details-container"
      >
        <h4
          className="product-card-title"
        >
          Nutri Well Barly
           
           
        </h4>
        <div
          className="details-image-container"
        >
          <img
            alt={120}
            src="https://4.bp.blogspot.com/-aI_KMIohZHQ/VN_63b-lsQI/AAAAAAAAZd4/6sifxckbEgQ/s1600/F%26N%2BNutriTea%2BBarley%2BDrink_1L.jpg"
            style={
              Object {
                "height": "200px",
              }
            }
          />
        </div>
        <div
          className="details-content-container"
        >
          <div
            className="details-content-container-capacity"
          >
            1
            L
          </div>
          <div
            className="details-content-container-price"
          >
            $
            2.25
          </div>
          <div
            className="details-content-container-description"
          >
            F&N NutriWell Barley is freshly brewed from a special home recipe using carefully selected pearl barley and dried winter melon strips. Barley, a grain full of pure goodness, is commonly used in home-brews, to make that familiar barley drink or in soups. All natural, with no added preservatives and reduced in sugar, F&N NutriWell Barley is the great-tasting
          </div>
          <button
            className="btn-flat btn-red"
            onClick={[Function]}
          >
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
